File: algorithms/warmup/cs/diagonal-difference.cs
static int diagonalDifference(List<List<int>> arr)
{
    var length = arr.Count();
    int indexDiag1 = 0, indexDiag2 = length - 1;

    var diag1 = new List<int>();
    var diag2 = new List<int>();

    foreach (var list in arr)
    {
        diag1.Add(list.ElementAt(indexDiag1));
        diag2.Add(list.ElementAt(indexDiag2));

        indexDiag1++;
        indexDiag2--;
    }

    return Math.Abs(diag1.Sum() - diag2.Sum());
}

File: algorithms/warmup/js/diagonal-difference.js
function diagonalDifference(arr) {
  var length = arr.length;
  var indexDiag1 = 0;
  var indexDiag2 = length - 1;

  var diag1 = [];
  var diag2 = [];

  arr.forEach((list) => {
    diag1 = [...diag1, list[indexDiag1]];
    diag2 = [...diag2, list[indexDiag2]];

    indexDiag1++;
    indexDiag2--;
  });

  var sumDiag1 = diag1.reduce((a, b) => {
    return a + b;
  }, 0);

  var sumDiag2 = diag2.reduce((a, b) => {
    return a + b;
  }, 0);

  return Math.abs(sumDiag1 - sumDiag2);
}
